src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -25980,5 +25775,837 @@ public void Add(MyCollection c)
+                  IL_0004:  newobj     "System.Collections.Generic.List<int>..ctor()"
+                  IL_0009:  dup
+                  IL_000a:  ldloca.s   V_0
+                  IL_000c:  call       "int System.ReadOnlySpan<int>.Length.get"
+                  IL_0011:  ldloca.s   V_1
+                  IL_0013:  call       "int System.ReadOnlySpan<int>.Length.get"
+                  IL_0018:  add
+                  IL_0019:  call       "void System.Runtime.InteropServices.CollectionsMarshal.SetCount<int>(System.Collections.Generic.List<int>, int)"
+                  IL_001e:  dup
+                  IL_001f:  call       "System.Span<int> System.Runtime.InteropServices.CollectionsMarshal.AsSpan<int>(System.Collections.Generic.List<int>)"
+                  IL_0024:  stloc.2
+                  IL_0025:  ldc.i4.0
+                  IL_0026:  stloc.3
+                  IL_0027:  ldloca.s   V_0
+                  IL_0029:  ldloca.s   V_2
+                  IL_002b:  ldloc.3
+                  IL_002c:  ldloca.s   V_0
+                  IL_002e:  call       "int System.ReadOnlySpan<int>.Length.get"
+                  IL_0033:  call       "System.Span<int> System.Span<int>.Slice(int, int)"
+                  IL_0038:  call       "void System.ReadOnlySpan<int>.CopyTo(System.Span<int>)"
+                  IL_003d:  ldloc.3
+                  IL_003e:  ldloca.s   V_0
+                  IL_0040:  call       "int System.ReadOnlySpan<int>.Length.get"
+                  IL_0045:  add
+                  IL_0046:  stloc.3
+                  IL_0047:  ldloca.s   V_1
+                  IL_0049:  ldloca.s   V_2
+                  IL_004b:  ldloc.3
+                  IL_004c:  ldloca.s   V_1
+                  IL_004e:  call       "int System.ReadOnlySpan<int>.Length.get"
+                  IL_0053:  call       "System.Span<int> System.Span<int>.Slice(int, int)"
+                  IL_0058:  call       "void System.ReadOnlySpan<int>.CopyTo(System.Span<int>)"
+                  IL_005d:  ldloc.3
+                  IL_005e:  ldloca.s   V_1
+                  IL_0060:  call       "int System.ReadOnlySpan<int>.Length.get"
+                  IL_0065:  add
+                  IL_0066:  stloc.3
+                  IL_0067:  ldc.i4.0
+                  IL_0068:  call       "void CollectionExtensions.Report(object, bool)"
+                  IL_006d:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void ArrayToList_Spreads()
+        {
+            var source = """
+                using System.Collections.Generic;
+
+                class C
+                {
+                    static void Main()
+                    {
+                        M([1, 2, 3], [4, 5, 6]);
+                    }
+
+                    static void M(int[] e1, int[] e2)
+                    {
+                        List<int> result = [..e1, ..e2];
+                        result.Report();
+                    }
+                }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

Consider testing spreading a multi-dimensional array, to verify we're skipping the optimization.